flowredux/src/commonMain/kotlin/com/freeletics/flowredux/dsl/ChangeState.kt

```diff
@@ -1,45 +1,15 @@
 package com.freeletics.flowredux.dsl
 
 /**
- * Represents a state transition.
- * Either use
- *  - [MutateState] to mutate the current state. You get a lambda block with the current state as input and expects
- *  the next state as output.
- * - [OverrideState] to trigger a transition from one state to the next one by simply overriding whatever state was
- * there before. Be careful using this especially if you have handle multiple actions or collect flows within the same
- * state as it really overrides the full state and you may loose information or partial state changes triggered by them.
- * Use [MutateState] in that case.
- * - [NoStateChange] to indicate that this function did not change the state at all. There should be only very little
- * use case for [NoStateChange]. So only use it if you actually know what you do.
+ * Represents a state transition. Instances of this a created through the [State.mutate],
+ * [State.override] and [State.noChange] methods.
  *
  * [ChangeState] doesnt allow you to directly access any property.
  * Then you may wonder how do you write unit test for one of your functions that return a [ChangeState]?
  * You need to call [ChangeState.reduce] to get the actual result of the change state.
  */
 public sealed class ChangeState<out S>
```

Comment:
Unrelated to this PR, but I am wondering if we should name this `ChangedState`?

My reasoning is then the method signatures of handlers reads more fluent:
```fun
fun onFooAction(action : Foo, state : State<MyState<) : ChangedState { ... }
```

So you  read it "I get State and return a changed state". 

What do you think? 
Or do you think we are  moving fast to remove `ChangeState` completely and have `State<MyState>` also as return type instead of `ChangedState`? I haven't spend much time yet to think about it ...

Reply:
ChangedState sounds good, I can update this here.

> Or do you think we are moving fast to remove ChangeState completely and have State<MyState> also as return type instead of ChangedState? I haven't spend much time yet to think about it ...

I also haven't though to much about whether using State everywhere could have some unintended downsides or side effects

Reply:
Then let's go with ChangedState for now